### PR TITLE
enable code executor by default for image tags >=3.20.15

### PR DESCRIPTION
**Pull request by avimoondra**
- Enables code executor by default for image tags >=3.20.15 (where we started publishing matching tags for CE consistently)
- Corresponds to docs changes here: https://github.com/tryretool/docs/pull/568
- Corresponds to other docs changes here: https://github.com/tryretool/docs/pull/572 that request 3.7+ https://docs.retool.com/self-hosted/quickstarts/kubernetes/helm#requirements. I tried with 3.3... template is broken.

### Helm 3.7 
we've had issues with this before: https://retooled.slack.com/archives/C03THV8QH0S/p1702051476282189

#### Inputs

```
echo "\n\n\n ====== Running tests for code-executor (helm 3.7) ======";
echo helm-3-7 version

echo "TEST: --set codeExecutor.enabled=true --set codeExecutor.image.tag=1.1.0";
helm-3-7 template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="5.6.10" --set codeExecutor.enabled=true --set codeExecutor.image.tag=1.1.0 | grep code-executor;
echo "expecting CE";
echo "\n";

echo "TEST: --set codeExecutor.enabled='true' --set codeExecutor.image.tag=1.1.0";
helm-3-7 template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="5.6.10" --set codeExecutor.enabled="true" --set codeExecutor.image.tag=1.1.0 | grep code-executor;
echo "expecting CE";
echo "\n";

echo "TEST: --set codeExecutor.enabled=false";
helm-3-7 template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="5.6.10" --set codeExecutor.enabled=false | grep code-executor;
echo "No CE";
echo "\n";

echo "TEST: --set codeExecutor.enabled='false'";
helm-3-7 template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="5.6.10" --set codeExecutor.enabled="false" | grep code-executor;
echo "No CE";
echo "\n";

echo "TEST: --set codeExecutor.image.tag=3.20.15";
helm-3-7 template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="5.6.10" --set codeExecutor.image.tag=3.20.15 | grep code-executor;
echo "expecting CE";
echo "\n";

echo "TEST: --set codeExecutor.image.tag=3.21.1";
helm-3-7 template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="5.6.10" --set codeExecutor.image.tag=3.21.1 | grep code-executor;
echo "expecting CE";
echo "\n";

echo "TEST: --set codeExecutor.image.tag=1.1.0";
helm-3-7 template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="5.6.10" --set codeExecutor.image.tag=1.1.0 | grep code-executor;
echo "No CE";
echo "\n";


echo "\n\n\n ====== Running tests for workflows (helm 3.7) ======";

echo "TEST: --set workflows.enabled=true ";
helm-3-7 template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="5.6.10" --set workflows.enabled=true  | grep worker;
echo "expecting worker";
echo "\n";

echo "TEST: --set workflows.enabled='true'";
helm-3-7 template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="5.6.10" --set workflows.enabled="true"  | grep worker;
echo "expecting worker";
echo "\n";

echo "TEST: --set workflows.enabled=false";
helm-3-7 template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="5.6.10" --set workflows.enabled=false | grep worker;
echo "No worker";
echo "\n";

echo "TEST: --set workflows.enabled='false'";
helm-3-7 template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="5.6.10" --set workflows.enabled="false" | grep worker;
echo "No worker";
echo "\n";

echo "TEST: --set image.tag=3.6.11";
helm-3-7 template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo"  --set image.tag=3.6.11 | grep worker;
echo "expecting worker";
echo "\n";

echo "TEST: --set image.tag=3.21.1";
helm-3-7 template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag=3.21.1 | grep worker;
echo "expecting worker";
echo "\n";

echo "TEST: --set image.tag=3.4.14";
helm-3-7 template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="3.4.14" | grep worker;
echo "No worker";
echo "\n";

echo "\n\n\n ====== Running tests for workflows w/ code executor (default) (helm 3.7) ======";

echo "TEST: --set image.tag=3.4.14 --set codeExecutor.image.tag=3.20.15";
helm-3-7 template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="3.20.16" --set codeExecutor.image.tag="3.20.16" | grep 'worker\|code-executor';
echo "worker AND code executor";
echo "\n";
```

#### Outputs

```
 ====== Running tests for code-executor (helm 3.7) ======
helm-3-7 version
TEST: --set codeExecutor.enabled=true --set codeExecutor.image.tag=1.1.0
  name: foo-retool-code-executor
    retoolService: foo-retool-code-executor
            value: http://foo-retool-code-executor
  name: foo-retool-code-executor
    retoolService: foo-retool-code-executor
      retoolService: foo-retool-code-executor
        prometheus.io/job: foo-retool-code-executor
        retoolService: foo-retool-code-executor
        image: "tryretool/code-executor-service:1.1.0"
            value: http://foo-retool-code-executor
            value: http://foo-retool-code-executor
expecting CE


TEST: --set codeExecutor.enabled='true' --set codeExecutor.image.tag=1.1.0
  name: foo-retool-code-executor
    retoolService: foo-retool-code-executor
            value: http://foo-retool-code-executor
  name: foo-retool-code-executor
    retoolService: foo-retool-code-executor
      retoolService: foo-retool-code-executor
        prometheus.io/job: foo-retool-code-executor
        retoolService: foo-retool-code-executor
        image: "tryretool/code-executor-service:1.1.0"
            value: http://foo-retool-code-executor
            value: http://foo-retool-code-executor
expecting CE


TEST: --set codeExecutor.enabled=false
No CE


TEST: --set codeExecutor.enabled='false'
No CE


TEST: --set codeExecutor.image.tag=3.20.15
  name: foo-retool-code-executor
    retoolService: foo-retool-code-executor
            value: http://foo-retool-code-executor
  name: foo-retool-code-executor
    retoolService: foo-retool-code-executor
      retoolService: foo-retool-code-executor
        prometheus.io/job: foo-retool-code-executor
        retoolService: foo-retool-code-executor
        image: "tryretool/code-executor-service:3.20.15"
            value: http://foo-retool-code-executor
expecting CE


TEST: --set codeExecutor.image.tag=3.21.1
  name: foo-retool-code-executor
    retoolService: foo-retool-code-executor
            value: http://foo-retool-code-executor
  name: foo-retool-code-executor
    retoolService: foo-retool-code-executor
      retoolService: foo-retool-code-executor
        prometheus.io/job: foo-retool-code-executor
        retoolService: foo-retool-code-executor
        image: "tryretool/code-executor-service:3.21.1"
            value: http://foo-retool-code-executor
expecting CE


TEST: --set codeExecutor.image.tag=1.1.0
No CE





 ====== Running tests for workflows (helm 3.7) ======
TEST: --set workflows.enabled=true
# Source: retool/templates/deployment_workflows_worker.yaml
  name: foo-retool-workflow-worker
    retoolService: foo-retool-workflow-worker
# Source: retool/templates/deployment_workflows_worker.yaml
  name: foo-retool-workflow-worker
    retoolService: foo-retool-workflow-worker
      retoolService: foo-retool-workflow-worker
        prometheus.io/job: foo-retool-workflow-worker
        retoolService: foo-retool-workflow-worker
expecting worker


TEST: --set workflows.enabled='true'
# Source: retool/templates/deployment_workflows_worker.yaml
  name: foo-retool-workflow-worker
    retoolService: foo-retool-workflow-worker
# Source: retool/templates/deployment_workflows_worker.yaml
  name: foo-retool-workflow-worker
    retoolService: foo-retool-workflow-worker
      retoolService: foo-retool-workflow-worker
        prometheus.io/job: foo-retool-workflow-worker
        retoolService: foo-retool-workflow-worker
expecting worker


TEST: --set workflows.enabled=false
No worker


TEST: --set workflows.enabled='false'
No worker


TEST: --set image.tag=3.6.11
# Source: retool/templates/deployment_workflows_worker.yaml
  name: foo-retool-workflow-worker
    retoolService: foo-retool-workflow-worker
# Source: retool/templates/deployment_workflows_worker.yaml
  name: foo-retool-workflow-worker
    retoolService: foo-retool-workflow-worker
      retoolService: foo-retool-workflow-worker
        prometheus.io/job: foo-retool-workflow-worker
        retoolService: foo-retool-workflow-worker
expecting worker


TEST: --set image.tag=3.21.1
# Source: retool/templates/deployment_workflows_worker.yaml
  name: foo-retool-workflow-worker
    retoolService: foo-retool-workflow-worker
# Source: retool/templates/deployment_workflows_worker.yaml
  name: foo-retool-workflow-worker
    retoolService: foo-retool-workflow-worker
      retoolService: foo-retool-workflow-worker
        prometheus.io/job: foo-retool-workflow-worker
        retoolService: foo-retool-workflow-worker
expecting worker


TEST: --set image.tag=3.4.14
No worker





 ====== Running tests for workflows w/ code executor (default) (helm 3.7) ======
TEST: --set image.tag=3.4.14 --set codeExecutor.image.tag=3.20.15
  name: foo-retool-code-executor
    retoolService: foo-retool-code-executor
# Source: retool/templates/deployment_workflows_worker.yaml
  name: foo-retool-workflow-worker
    retoolService: foo-retool-workflow-worker
            value: http://foo-retool-code-executor
  name: foo-retool-code-executor
    retoolService: foo-retool-code-executor
      retoolService: foo-retool-code-executor
        prometheus.io/job: foo-retool-code-executor
        retoolService: foo-retool-code-executor
        image: "tryretool/code-executor-service:3.20.16"
# Source: retool/templates/deployment_workflows_worker.yaml
  name: foo-retool-workflow-worker
    retoolService: foo-retool-workflow-worker
      retoolService: foo-retool-workflow-worker
        prometheus.io/job: foo-retool-workflow-worker
        retoolService: foo-retool-workflow-worker
            value: http://foo-retool-code-executor
worker AND code executor
```

#### Inputs

```
echo "\n\n\n ====== Running tests for code-executor (helm) ======";
helm version

echo "TEST: --set codeExecutor.enabled=true --set codeExecutor.image.tag=1.1.0";
helm template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="5.6.10" --set codeExecutor.enabled=true --set codeExecutor.image.tag=1.1.0 | grep code-executor;
echo "expecting CE";
echo "\n";

echo "TEST: --set codeExecutor.enabled='true' --set codeExecutor.image.tag=1.1.0";
helm template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="5.6.10" --set codeExecutor.enabled="true" --set codeExecutor.image.tag=1.1.0 | grep code-executor;
echo "expecting CE";
echo "\n";

echo "TEST: --set codeExecutor.enabled=false";
helm template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="5.6.10" --set codeExecutor.enabled=false | grep code-executor;
echo "No CE";
echo "\n";

echo "TEST: --set codeExecutor.enabled='false'";
helm template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="5.6.10" --set codeExecutor.enabled="false" | grep code-executor;
echo "No CE";
echo "\n";

echo "TEST: --set codeExecutor.image.tag=3.20.15";
helm template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="5.6.10" --set codeExecutor.image.tag=3.20.15 | grep code-executor;
echo "expecting CE";
echo "\n";

echo "TEST: --set codeExecutor.image.tag=3.21.1";
helm template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="5.6.10" --set codeExecutor.image.tag=3.21.1 | grep code-executor;
echo "expecting CE";
echo "\n";

echo "TEST: --set codeExecutor.image.tag=1.1.0";
helm template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="5.6.10" --set codeExecutor.image.tag=1.1.0 | grep code-executor;
echo "No CE";
echo "\n";


echo "\n\n\n ====== Running tests for workflows (helm) ======";

echo "TEST: --set workflows.enabled=true ";
helm template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="5.6.10" --set workflows.enabled=true  | grep worker;
echo "expecting worker";
echo "\n";

echo "TEST: --set workflows.enabled='true'";
helm template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="5.6.10" --set workflows.enabled="true"  | grep worker;
echo "expecting worker";
echo "\n";

echo "TEST: --set workflows.enabled=false";
helm template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="5.6.10" --set workflows.enabled=false | grep worker;
echo "No worker";
echo "\n";

echo "TEST: --set workflows.enabled='false'";
helm template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="5.6.10" --set workflows.enabled="false" | grep worker;
echo "No worker";
echo "\n";

echo "TEST: --set image.tag=3.6.11";
helm template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo"  --set image.tag=3.6.11 | grep worker;
echo "expecting worker";
echo "\n";

echo "TEST: --set image.tag=3.21.1";
helm template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag=3.21.1 | grep worker;
echo "expecting worker";
echo "\n";

echo "TEST: --set image.tag=3.4.14";
helm template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="3.4.14" | grep worker;
echo "No worker";
echo "\n";

echo "\n\n\n ====== Running tests for workflows w/ code executor (default) (helm) ======";

echo "TEST: --set image.tag=3.4.14 --set codeExecutor.image.tag=3.20.15";
helm template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="3.20.16" --set codeExecutor.image.tag="3.20.16" | grep 'worker\|code-executor';
echo "worker AND code executor";
echo "\n";

```

#### Outputs

```
 ====== Running tests for code-executor (helm) ======
version.BuildInfo{Version:"v3.10.3", GitCommit:"835b7334cfe2e5e27870ab3ed4135f136eecc704", GitTreeState:"clean", GoVersion:"go1.19.4"}
TEST: --set codeExecutor.enabled=true --set codeExecutor.image.tag=1.1.0
  name: foo-retool-code-executor
    retoolService: foo-retool-code-executor
            value: http://foo-retool-code-executor
  name: foo-retool-code-executor
    retoolService: foo-retool-code-executor
      retoolService: foo-retool-code-executor
        prometheus.io/job: foo-retool-code-executor
        retoolService: foo-retool-code-executor
        image: "tryretool/code-executor-service:1.1.0"
            value: http://foo-retool-code-executor
            value: http://foo-retool-code-executor
expecting CE


TEST: --set codeExecutor.enabled='true' --set codeExecutor.image.tag=1.1.0
  name: foo-retool-code-executor
    retoolService: foo-retool-code-executor
            value: http://foo-retool-code-executor
  name: foo-retool-code-executor
    retoolService: foo-retool-code-executor
      retoolService: foo-retool-code-executor
        prometheus.io/job: foo-retool-code-executor
        retoolService: foo-retool-code-executor
        image: "tryretool/code-executor-service:1.1.0"
            value: http://foo-retool-code-executor
            value: http://foo-retool-code-executor
expecting CE


TEST: --set codeExecutor.enabled=false
No CE


TEST: --set codeExecutor.enabled='false'
No CE


TEST: --set codeExecutor.image.tag=3.20.15
  name: foo-retool-code-executor
    retoolService: foo-retool-code-executor
            value: http://foo-retool-code-executor
  name: foo-retool-code-executor
    retoolService: foo-retool-code-executor
      retoolService: foo-retool-code-executor
        prometheus.io/job: foo-retool-code-executor
        retoolService: foo-retool-code-executor
        image: "tryretool/code-executor-service:3.20.15"
            value: http://foo-retool-code-executor
expecting CE


TEST: --set codeExecutor.image.tag=3.21.1
  name: foo-retool-code-executor
    retoolService: foo-retool-code-executor
            value: http://foo-retool-code-executor
  name: foo-retool-code-executor
    retoolService: foo-retool-code-executor
      retoolService: foo-retool-code-executor
        prometheus.io/job: foo-retool-code-executor
        retoolService: foo-retool-code-executor
        image: "tryretool/code-executor-service:3.21.1"
            value: http://foo-retool-code-executor
expecting CE


TEST: --set codeExecutor.image.tag=1.1.0
No CE





 ====== Running tests for workflows (helm) ======
TEST: --set workflows.enabled=true
# Source: retool/templates/deployment_workflows_worker.yaml
  name: foo-retool-workflow-worker
    retoolService: foo-retool-workflow-worker
# Source: retool/templates/deployment_workflows_worker.yaml
  name: foo-retool-workflow-worker
    retoolService: foo-retool-workflow-worker
      retoolService: foo-retool-workflow-worker
        prometheus.io/job: foo-retool-workflow-worker
        retoolService: foo-retool-workflow-worker
expecting worker


TEST: --set workflows.enabled='true'
# Source: retool/templates/deployment_workflows_worker.yaml
  name: foo-retool-workflow-worker
    retoolService: foo-retool-workflow-worker
# Source: retool/templates/deployment_workflows_worker.yaml
  name: foo-retool-workflow-worker
    retoolService: foo-retool-workflow-worker
      retoolService: foo-retool-workflow-worker
        prometheus.io/job: foo-retool-workflow-worker
        retoolService: foo-retool-workflow-worker
expecting worker


TEST: --set workflows.enabled=false
No worker


TEST: --set workflows.enabled='false'
No worker


TEST: --set image.tag=3.6.11
# Source: retool/templates/deployment_workflows_worker.yaml
  name: foo-retool-workflow-worker
    retoolService: foo-retool-workflow-worker
# Source: retool/templates/deployment_workflows_worker.yaml
  name: foo-retool-workflow-worker
    retoolService: foo-retool-workflow-worker
      retoolService: foo-retool-workflow-worker
        prometheus.io/job: foo-retool-workflow-worker
        retoolService: foo-retool-workflow-worker
expecting worker


TEST: --set image.tag=3.21.1
# Source: retool/templates/deployment_workflows_worker.yaml
  name: foo-retool-workflow-worker
    retoolService: foo-retool-workflow-worker
# Source: retool/templates/deployment_workflows_worker.yaml
  name: foo-retool-workflow-worker
    retoolService: foo-retool-workflow-worker
      retoolService: foo-retool-workflow-worker
        prometheus.io/job: foo-retool-workflow-worker
        retoolService: foo-retool-workflow-worker
expecting worker


TEST: --set image.tag=3.4.14
No worker


 ====== Running tests for workflows w/ code executor (default) (helm) ======
TEST: --set image.tag=3.4.14 --set codeExecutor.image.tag=3.20.15
  name: foo-retool-code-executor
    retoolService: foo-retool-code-executor
# Source: retool/templates/deployment_workflows_worker.yaml
  name: foo-retool-workflow-worker
    retoolService: foo-retool-workflow-worker
            value: http://foo-retool-code-executor
  name: foo-retool-code-executor
    retoolService: foo-retool-code-executor
      retoolService: foo-retool-code-executor
        prometheus.io/job: foo-retool-code-executor
        retoolService: foo-retool-code-executor
        image: "tryretool/code-executor-service:3.20.16"
# Source: retool/templates/deployment_workflows_worker.yaml
  name: foo-retool-workflow-worker
    retoolService: foo-retool-workflow-worker
      retoolService: foo-retool-workflow-worker
        prometheus.io/job: foo-retool-workflow-worker
        retoolService: foo-retool-workflow-worker
            value: http://foo-retool-code-executor
worker AND code executor
```

### CODE REVIEW

File: charts/retool/templates/_helpers.tpl [hunk 1]
```diff
@@ -141,13 +141,11 @@ Usage: (include "retool.workflows.enabled" .)
 */}}
 {{- define "retool.workflows.enabled" -}}
 {{- $output := "" -}}
-{{- if or 
-    (eq (toString (default "" .Values.workflows.enabled)) "true")
-    (eq .Values.workflows.enabled true) 
-    (eq (toString (default "" .Values.workflows.enabled)) "false") 
-    (eq .Values.workflows.enabled false)
```
Comment (avimoondra) [inline charts/retool/templates/_helpers.tpl:148]: on main, with 3.7 I was getting errors like: 

```
Error: template: retool/templates/deployment_workflows_worker.yaml:1:7: executing "retool/templates/deployment_workflows_worker.yaml" at <include "retool.workflows.enabled" .>: error calling include: template: retool/templates/_helpers.tpl:146:5: executing "retool.workflows.enabled" at <eq .Values.workflows.enabled true>: error calling eq: incompatible types for comparison
```

when not explicitly setting workflows.enabled.

File: charts/retool/templates/_helpers.tpl [hunk 2]
```diff
@@ -141,13 +141,11 @@ Usage: (include "retool.workflows.enabled" .)
...
-    (eq (toString (default "" .Values.workflows.enabled)) "true")
-    (eq .Values.workflows.enabled true) 
-    (eq (toString (default "" .Values.workflows.enabled)) "false") 
-    (eq .Values.workflows.enabled false) 
+{{- if or
+    (eq (toString .Values.workflows.enabled) "true")
+    (eq (toString .Values.workflows.enabled) "false")
 -}}
-  {{- if eq .Values.workflows.enabled true -}}
+  {{- if (eq (toString .Values.workflows.enabled) "true") -}}
```
Comment (avimoondra) [inline charts/retool/templates/_helpers.tpl:148]: apply the same positive conditional as matched above

File: charts/retool/templates/_helpers.tpl [hunk 3]
```diff
@@ -164,6 +162,31 @@ Usage: (include "retool.workflows.enabled" .)
...
+    (eq (toString .Values.codeExecutor.enabled) "false")
+-}}
+  {{- if (eq (toString .Values.codeExecutor.enabled) "true") -}}
+    {{- $output = "1" -}}
+  {{- else -}}
+    {{- $output = "" -}}
+  {{- end -}}
+{{- else if empty .Values.codeExecutor.image.tag -}}
+  {{- $output = "" -}}
+{{- else if semverCompare ">= 3.20.15" .Values.codeExecutor.image.tag -}}
```
Comment (avimoondra) [inline charts/retool/templates/_helpers.tpl:182]: 3.20.15 is the latest patch of 3.20, which is when we started shipping code executors with matching tags consistently

File: charts/retool/templates/_helpers.tpl (hunk 1 above)
Comment (anna-yn) [reply charts/retool/templates/_helpers.tpl:148]: no need for the default anymore?

Comment (avimoondra) [reply charts/retool/templates/_helpers.tpl:148]: (this has been broken for a little while I think)

See history here... 
- https://github.com/tryretool/retool-helm/pull/120
- https://github.com/tryretool/retool-helm/pull/134
- https://github.com/tryretool/retool-helm/pull/135

https://retooled.slack.com/archives/C03THV8QH0S/p1702051476282189
-

Comment (avimoondra) [reply charts/retool/templates/_helpers.tpl:148]: seems like no @anna-yn 

```
helm-3-7 template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="3.20.16" --set codeExecutor.image.tag="3.20.16" | grep 'worker\|code-executor';
  name: foo-retool-code-executor
    retoolService: foo-retool-code-executor
# Source: retool/templates/deployment_workflows_worker.yaml
  name: foo-retool-workflow-worker
    retoolService: foo-retool-workflow-worker
            value: http://foo-retool-code-executor
  name: foo-retool-code-executor
    retoolService: foo-retool-code-executor
      retoolService: foo-retool-code-executor
        prometheus.io/job: foo-retool-code-executor
        retoolService: foo-retool-code-executor
        image: "tryretool/code-executor-service:3.20.16"
# Source: retool/templates/deployment_workflows_worker.yaml
  name: foo-retool-workflow-worker
    retoolService: foo-retool-workflow-worker
      retoolService: foo-retool-workflow-worker
        prometheus.io/job: foo-retool-workflow-worker
        retoolService: foo-retool-workflow-worker
            value: http://foo-retool-code-executor
```

in this case we are using the values.yml file in this repo, on this branch, where workflows.enabled nor codeExecutor.enabled is NOT set (blank) ... and we are using the image tags to determine if workflows should spin up or not.

**Review by anna-yn — APPROVED** [review]

File: charts/retool/templates/_helpers.tpl (hunk 1 above)
Comment (avimoondra) [reply charts/retool/templates/_helpers.tpl:148]: here's the same example from main branch... 

```
retool-helm git:(main) ✗ helm-3-7 template -f ~/retool-helm/charts/retool/values.yaml foo ~/retool-helm/charts/retool --set config.encryptionKey="foo" --set image.tag="3.20.16" --set codeExecutor.image.tag="3.20.16" | grep 'worker\|code-executor';
Error: template: retool/templates/deployment_workflows_worker.yaml:1:7: executing "retool/templates/deployment_workflows_worker.yaml" at <include "retool.workflows.enabled" .>: error calling include: template: retool/templates/_helpers.tpl:146:5: executing "retool.workflows.enabled" at <eq .Values.workflows.enabled true>: error calling eq: incompatible types for comparison

Use --debug flag to render out invalid YAML
```

File: charts/retool/templates/_helpers.tpl (hunk 3 above)
Comment (tanaythan) [reply charts/retool/templates/_helpers.tpl:182]: Is there a way we can automatically set `codeExecutor.image.tag` to the retool version? Could simplify things later on

**Review by tanaythan — APPROVED** [review]

File: charts/retool/templates/_helpers.tpl (hunk 3 above)
Comment (avimoondra) [reply charts/retool/templates/_helpers.tpl:182]: I was thinking that as well, will follow up with that change next

**Pull request merged**